### PR TITLE
Create UI Main

### PR DESCRIPTION
**Pull request by NguyenNgocTuyenUED**
Create UI Main Screen
Link: https://edu-redmine.sun-asterisk.vn/issues/8855
image

### CODE REVIEW

**Comment by NguyenNgocTuyenUED** [comment]
ready

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -9,7 +9,9 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/AppTheme" >
-        <activity android:name=".MainActivity" >
+        <activity android:name=".View.MainActivity"
+            android:label="@string/app_name"
```
Comment (thaohq-0814) [inline app/src/main/AndroidManifest.xml:13]: add screen orientation

File: app/src/main/java/com/sun_asterisk/youtubebackground/View/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.Toolbar;
+import android.view.Menu;
+import android.view.MenuItem;
+
+import com.sun_asterisk.youtubebackground.R;
+
+import static android.support.v7.widget.RecyclerView.*;
+
+
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/youtubebackground/View/MainActivity.java]: remove blank

File: app/src/main/java/com/sun_asterisk/youtubebackground/View/MainActivity.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        return super.onOptionsItemSelected(item);
+    }
+
+    @SuppressWarnings("StatementWithEmptyBody")
+    @Override
+    public boolean onNavigationItemSelected(MenuItem item) {
+        // Handle navigation view item clicks here.
+        int id = item.getItemId();
+
+        if (id == R.id.nav_home) {
```
Comment (thaohq-0814) [inline app/src/main/java/com/sun_asterisk/youtubebackground/View/MainActivity.java]: use switch()

File: app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -1,19 +1,26 @@
...
+    tools:openDrawer="start">
+
+    <include
+        layout="@layout/app_bar_main"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent" />
 
-    <TextView
+    <android.support.design.widget.NavigationView
+        android:id="@+id/nav_view"
```
Comment (thaohq-0814) [inline app/src/main/res/layout/activity_main.xml]:
```suggestion
        android:id="@+id/navigationMain"
```

File: app/src/main/res/layout/content_main.xml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    app:layout_behavior="@string/appbar_scrolling_view_behavior"
+    tools:context=".View.MainActivity"
+    tools:showIn="@layout/app_bar_main">
+
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/rvVideo"
```
Comment (thaohq-0814) [inline app/src/main/res/layout/content_main.xml]:
```suggestion
        android:id="@+id/recyclerViewVideo"
```

File: app/src/main/res/layout/nav_header_main.xml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    android:gravity="bottom"
+    android:orientation="vertical"
+    android:paddingLeft="@dimen/dp_16"
+    android:paddingTop="@dimen/dp_16"
+    android:paddingRight="@dimen/dp_16"
+    android:paddingBottom="@dimen/dp_16"
+    android:theme="@style/ThemeOverlay.AppCompat.Dark">
+
+    <ImageView
+        android:id="@+id/imageView"
```
Comment (thaohq-0814) [inline app/src/main/res/layout/nav_header_main.xml]:
```suggestion
        android:id="@+id/imageViewAvatar"
```
Đặt id cho có nghĩa bạn ơi.

File: app/src/main/res/layout/nav_header_main.xml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+    <TextView
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="@string/nav_header_title"
+        android:paddingTop="@dimen/dp_8"
+        android:textAppearance="@style/TextAppearance.AppCompat.Body1" />
+
+    <TextView
+        android:id="@+id/textView"
```
Comment (thaohq-0814) [inline app/src/main/res/layout/nav_header_main.xml]: as above

File: app/src/main/res/menu/activity_main_drawer.xml [hunk 8]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    tools:showIn="navigation_view">
+
+    <group android:checkableBehavior="single">
+        <item
+            android:id="@+id/nav_home"
```
Comment (thaohq-0814) [inline app/src/main/res/menu/activity_main_drawer.xml]:
```suggestion
            android:id="@+id/navigationHome"
```

File: app/src/main/res/menu/activity_main_drawer.xml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+            android:icon="@drawable/ic_home"
+            android:title="Home" />
+        <item
+            android:id="@+id/nav_history"
```
Comment (thaohq-0814) [inline app/src/main/res/menu/activity_main_drawer.xml]: as above

File: app/src/main/res/menu/activity_main_drawer.xml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+            android:icon="@drawable/ic_history"
+            android:title="History" />
+        <item
+            android:id="@+id/nav_setting"
```
Comment (thaohq-0814) [inline app/src/main/res/menu/activity_main_drawer.xml]: as above

File: app/src/main/res/menu/activity_main_drawer.xml [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+            android:icon="@drawable/ic_settings"
+            android:title="Setting" />
+    </group>
+
+    <item android:title="Communicate">
+        <menu>
+            <item
+                android:id="@+id/nav_share"
```
Comment (thaohq-0814) [inline app/src/main/res/menu/activity_main_drawer.xml]: as above

File: app/src/main/res/menu/activity_main_drawer.xml [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+                android:icon="@drawable/ic_share"
+                android:title="Share" />
+            <item
+                android:id="@+id/nav_introduce"
```
Comment (thaohq-0814) [inline app/src/main/res/menu/activity_main_drawer.xml]: as above

File: app/src/main/res/menu/main.xml [hunk 13]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <item
+        android:id="@+id/action_searchs"
```
Comment (thaohq-0814) [inline app/src/main/res/menu/main.xml]:
```suggestion
        android:id="@+id/actionSearch"
```

File: app/src/main/res/values/strings.xml [hunk 14]
```diff
@@ -1,3 +1,12 @@
 <resources>
     <string name="app_name">Youtube Background</string>
+    <string name="navigation_drawer_open">Open navigation drawer</string>
+    <string name="navigation_drawer_close">Close navigation drawer</string>
+    <string name="nav_header_title">Sun Asterisk</string>
+    <string name="nav_header_subtitle">sun-asterisk.com</string>
+    <string name="nav_header_desc">Navigation header</string>
+    <string name="action_searchs">Searchs</string>
+
```
Comment (thaohq-0814) [inline app/src/main/res/values/strings.xml]: remove blank

File: app/src/main/res/layout/nav_header_main.xml [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="@dimen/dp_176"
+    android:background="@drawable/side_nav_bar"
+    android:gravity="bottom"
+    android:orientation="vertical"
+    android:paddingLeft="@dimen/dp_16"
+    android:paddingTop="@dimen/dp_16"
+    android:paddingRight="@dimen/dp_16"
+    android:paddingBottom="@dimen/dp_16"
```
Comment (QuangHieuu) [inline app/src/main/res/layout/nav_header_main.xml]:
```
    android:paddingLeft="@dimen/dp_16"
    android:paddingTop="@dimen/dp_16"
    android:paddingRight="@dimen/dp_16"
    android:paddingBottom="@dimen/dp_16"
```

-> android:padding="@dimen/dp_16"

**Review by thaohq-0814 — APPROVED** [review]

**Comment by NguyenNgocTuyenUED** [comment]
ready

File: app/src/main/java/com/sun_asterisk/youtubebackground/View/MainActivity.java [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+
+import static android.support.v7.widget.RecyclerView.LayoutManager;
+
+public class MainActivity extends AppCompatActivity
+        implements NavigationView.OnNavigationItemSelectedListener {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/youtubebackground/View/MainActivity.java:22]: tách method ra e. chia từng method với từng chức năng cụ thể

File: app/src/main/java/com/sun_asterisk/youtubebackground/View/MainActivity.java [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+        ActionBarDrawerToggle toggle =
+                new ActionBarDrawerToggle(this, drawer, toolbar, R.string.navigation_drawer_open,
+                        R.string.navigation_drawer_close);
+        drawer.addDrawerListener(toggle);
+        toggle.syncState();
+        NavigationView navigationView = (NavigationView) findViewById(R.id.navigationMain);
+        navigationView.setNavigationItemSelectedListener(this);
+        RecyclerView rvVideo = findViewById(R.id.recycleViewVideo);
+        LayoutManager layoutManager = new LinearLayoutManager(this);
+        rvVideo.setLayoutManager(layoutManager);
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/youtubebackground/View/MainActivity.java]:
```suggestion
        rvVideo.setLayoutManager(new LinearLayoutManager(this));
```

File: app/src/main/res/layout/content_main.xml [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+    android:layout_height="match_parent"
+    app:layout_behavior="@string/appbar_scrolling_view_behavior"
+    tools:showIn="@layout/app_bar_main"
+    >
+
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/recycleViewVideo"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        >
```
Comment (tuanlvt-1117) [inline app/src/main/res/layout/content_main.xml]:
```suggestion
        />
```

**Comment by NguyenNgocTuyenUED** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**